runtime/src/bank.rs

```diff
@@ -30,27 +33,155 @@ use solana_sdk::timing::{duration_as_ms, duration_as_us, MAX_RECENT_BLOCKHASHES}
 use solana_sdk::transaction::{Result, Transaction, TransactionError};
 use std::borrow::Borrow;
 use std::cmp;
+use std::collections::HashMap;
+use std::fmt;
+use std::io::Cursor;
 use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
 use std::sync::{Arc, RwLock, RwLockReadGuard};
 use std::time::Instant;
 
 type BankStatusCache = StatusCache<Result<()>>;
 
-/// Manager for the state of all accounts and programs after processing its entries.
 #[derive(Default)]
-pub struct Bank {
+pub struct BankRc {
```

Comment:
what's the Rc suffix mean?

Reply:
> what's the Rc suffix mean?

This just holds the Arc members of the bank as they need to be serialized in a special manner and the test of the members of bank can be serialized directly.